cli/pkg/config/cli_config.go

```diff
@@ -43,11 +61,28 @@ func (c *CLIConfigManager) LoadCLIConfig() (CLIConfig, error) {
 		return cliConfig, nil
 	}
 
-	data, err := fileutils.ReadFile(c.CLIConfigPath)
-	if err != nil {
+	// Allow setting file path after CLIConfigManager is initialized
+	c.viper.SetConfigFile(c.CLIConfigPath)
+	if err := c.viper.ReadInConfig(); err != nil {
 		return cliConfig, fmt.Errorf("error when reading config file: %w", err)
 	}
-	if err := json.Unmarshal(data, &cliConfig); err != nil {
+
+	if err := c.viper.Unmarshal(&cliConfig, func(dConfig *mapstructure.DecoderConfig) {
+		dConfig.DecodeHook = mapstructure.StringToTimeHookFunc(time.RFC3339)
+	}); err != nil {
+		return cliConfig, fmt.Errorf("error when unmarshalling config file: %w", err)
+	}
+
+	return cliConfig, nil
+}
+
+// GetCLIConfig gets the already loaded configuration
+func (c *CLIConfigManager) GetCLIConfig() (CLIConfig, error) {
```

Comment:
Many places in the code use `LoadCLIConfig` which reads in the config file every time `LoadCLIConfig` function is called. `GetCLIConfig` is lighter version of the function. It only returns config already loaded in the Viper.